src/daemon/gnet/pool.go

```diff
@@ -424,11 +429,37 @@ func readLoop(conn *Connection, timeout time.Duration, maxMsgLen int, msgChan ch
 	}
 }
 
+func (pool *ConnectionPool) sendLoop(conn *Connection, timeout time.Duration) error {
+	for m := range conn.WriteQueue {
+		if m == nil {
+			continue
+		}
+
+		err := sendMessage(conn.Conn, m, timeout)
+		sr := newSendResult(conn.Addr(), m, err)
+		select {
+		case pool.SendResults <- sr:
+		case <-time.After(3 * time.Second):
+			logger.Warning("push send result channel timeout")
+		}
+
+		if err != nil {
+			return err
+		}
+
+		if err := pool.updateLastSent(conn.Addr(), Now()); err != nil {
```

Comment:
This updates last sent even if timed out

Reply:
The `timeout` here is caused by sending `SendResult` to daemon, the data actually have been sent if the err is nil, so it should update the last sent of this connection.